### PR TITLE
support operators in filters

### PR DESCRIPTION
**Pull request by G3z**
TL;DR;
Support for query operators like `->where('foo','>','1')` short (old) version still works `->where('foo','1')`

Both Algolia and MeiliSearch supports different operators when filtering:
` <, <=, =, !=, >=, > ` are available for both engine

[Algolia docs](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/how-to/filter-by-numeric-value/#applying-a-numeric-filter)
[MeiliSearch docs](https://docs.meilisearch.com/reference/features/filtering_and_faceted_search.html#using-filters)

this PR enables the user to choose a different operator from the default (`=`) while maintaining backward compatibility

solves #529

### CODE REVIEW

**Comment by driesvints** [comment]
@G3z please add a much more thorough explanation to the pr for Taylor to review or it will be closed.

**Comment by G3z** [comment]
i hope it's clearer now

**Comment by taylorotwell** [comment]
This would have to be part of a new major release because any third-party Scout drivers would not be expecting where clauses with this format.

**Comment by G3z** [comment]
should I re-target this on master to target the next major ?

**Comment by driesvints** [comment]
@G3z yeah best to target master
